prism/prism.c

```diff
@@ -21711,7 +21711,7 @@ pm_strnstr(const char *big, const char *little, size_t big_length) {
  */
 static void
 pm_parser_warn_shebang_carriage_return(pm_parser_t *parser, const uint8_t *start, size_t length) {
-    if (length > 2 && start[length - 1] == '\n' && start[length - 2] == '\r') {
+    if (length > 2 && start[length - 1] == '\r' && start[length] == '\n') {
```

Comment:
FIXME: out-of-bounds access https://github.com/ruby/prism/pull/3007#pullrequestreview-2257164857